_posts/2015-11-30-translation-process.md

```diff
@@ -0,0 +1,96 @@
+
+In this article I will not focus on why it is important to have good copies in your product to drive engagement, reflect your product’s quality and define your brand’s positioning, instead, I will talk about the tools and processes which helped us handle translations at Onefootball.
+
+
+
+## The actors - who is involved in the translation process?
+
+At Onefootball, the following actors are involved, successively:
+
+* **Feature owner** – he/she is the responsible of a new feature and its definition, new copies included.
+* **Native speaker** – we always have a native English speaker reviewing a new feature’s copies to sign them off. Ideally this person is part of the Marketing team and can also align the tone of voice of the final copies. 
+* **Translation manager** – in our set-up, this role is covered by Technical Product Managers. They are responsible for preparing the ground for translation requests (we’ll see that later) and for co-ordinating the translators. At last, they are also responsible for making sure translations collected are propagated correctly to all platforms.
+* **Translators** –they can either be internal or external resources to the company. The important is that they are native speakers.
+* **Developers** – they are the one implementing the new features and the new copies. They are responsible for their platform-specific translation file.
+
+
+
+## The process - how do those actors collaborate?
+
+The translation flow is compounded of three phases:
+
+* **Copy definition** – once the feature owner has defined the copies for the new feature, he passes them to the native speaker for sign-off. At the end of this step, the final copies are defined, reviewed and approved. They can be passed to the translation managers. English-UK is always used as a reference.
+* **Translation management** – once they receive new copies, the translation managers add them in the collaborative translation tool (which we’ll see later). From there, they can easily request translations to the translators, monitor the progress and access the final outputs. At this point, English-UK should have been declined in as many languages as your product supports.
+* **Integration** – developers are responsible for their translation file and should make sure to keep it updated on the collaborative translation tool. In our set-up each platform has its own project on our collaborative tool. Developers make sure to keep their project updated by uploading their platform’s translation file in English-UK  (AKA “root translation file”) to their project as soon as there are new copies. Then translation managers populate the translations collected earlier to platform-specific projects. At the end, developers can download the final translation files in one click.
```

Comment:
Ideally the download phase is somehow automated. As we discussed, I think it would be worth describing as well as naming the tools we use and giving concrete examples of the process on the developer side.

I the web team we have two grunt tasks, `translations:up` and `translations:down`, that we run respectively to upload the new keys added to the source file and download all the updated translation files. The tasks rely on the default crowdin client.

```
    //https://github.com/sindresorhus/grunt-shell
    shell: {
        upload: {
            command: 'cd tools/ && java -jar crowdin-cli.jar upload source ; cd -'
        },
        download: {
            command: 'cd tools/ && java -jar crowdin-cli.jar download translations; cd -'
        }
    },
    ...
    grunt.registerTask('translations:up', ['shell:upload']);
    grunt.registerTask('translations:down', ['shell:download']);
```